domehunter/dome_control.py

```diff
@@ -0,0 +1,674 @@
+            self._move_cw()
+            if self.testing:
+                # tell the fake home sensor that we have left home
+                self.home_sensor_pin.drive_low()
+                self._simulate_ticks(num_ticks=10)
+            # need to introduce a small pause before wait_for_active as we
+            # start from home position and want to make sure we clear the
+            # sensor before we start waiting for it to come around again
+            time.sleep(0.5)
+            self.home_sensor.wait_for_active(timeout=self.wait_timeout)
+
+            if self.testing:
+                # tell the fake home sensor that we have come back to home
+                self.home_sensor_pin.drive_high()
+
+            self._stop_moving()
+            # another short pause to get an obvious blink of debug_lights
+            time.sleep(0.1)
+
+            rotation_count += 1
+
+        # set the azimuth per encoder tick factor based on how many ticks we
+        # counted over n rotations
+        self.az_per_tick = 360 / (self.encoder_count / rotation_count)
+        # set dome azimuth to 0?
+        # problem, this isn't compensating for any overshooting
+        self.dome_az = 0
+
+    def find_home(self):
+        """
+        Move Dome to home position.
+
+        Parameters
+        ----------
+        num_cal_rotations : integer
+            Number of rotations to perform to calibrate encoder.
+
+        """
+        # if dome already home, do nothing
+        if self.is_home:
+            self.encoder_count = 0
+        else:
+            self._move_cw()
+            self.home_sensor.wait_for_active(timeout=self.wait_timeout)
+            if self.testing:
+                # in testing mode need to "fake" the activation of the home pin
+                self.home_sensor_pin.drive_high()
+
+            self._stop_moving()
+            self.encoder_count = 0
+
+###############################################################################
+# Private Methods
+###############################################################################
+
+    def _set_at_home(self):
+        """
+        Update home status to at home and debug LEDs (if enabled).
+        """
+        self._turn_led_on(leds=['input_2'])
+        self._at_home = True
+
+    def _set_not_home(self):
+        """
+        Update home status to not at home and debug LEDs (if enabled).
+        """
+        self._turn_led_off(leds=['input_2'])
+        self._at_home = False
+
+    def _increment_count(self):
+        """
+        Private method used for callback function of the encoder DOD.
+
+        Calling this method will toggle the encoder debug LED (if enabled)
+        and increment or decrement the encoder_count instance variable,
+        depending on the current rotation direction of the dome.
+
+        If the current dome direction cannot be determined, the last recorded
+        direction is adopted.
+        """
+        print(f"Encoder activated _increment_count")
```

Comment:
Probably don't want a `print` statement buried in here. Which makes me realize there should probably be a lot more logging going on.